framework/src/problems/FEProblemBase.C

```diff
@@ -912,16 +912,6 @@ FEProblemBase::initialSetup()
   if (_displaced_problem && _mortar_data.hasDisplacedObjects())
     _displaced_problem->updateMesh();
 
-  // Build the mortar segment meshes for a couple reasons:
-  // 1) Get the ghosting correct for both static and dynamic meshes
-  // 2) Make sure the mortar mesh is built for mortar constraints that live on the static mesh
-  //
-  // It is worth-while to note that mortar meshes that live on a dynamic mesh will be built
-  // during residual and Jacobian evaluation because when displacements are solution variables
-  // the mortar mesh will move and change during the course of a non-linear solve. We DO NOT
-  // redo ghosting during non-linear solve, so for purpose 1) the below call has to be made
-  updateMortarMesh();
```

Comment:
What's the implication of moving `updateMortarMesh()` from `FEProblemBase::initialSetup()` to `FEProblemBase::init()` ?